src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -286,26 +325,52 @@ std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_try_translate_expre
     case ExpressionType::Predicate:
     case ExpressionType::Arithmetic:
     case ExpressionType::Logical: {
-      std::vector<std::shared_ptr<const JitExpression>> jit_expression_arguments;
+      const bool use_value_ids = can_use_value_ids_in_expression(expression);
+
+      std::vector<std::shared_ptr<JitExpression>> jit_expression_arguments;
       for (const auto& argument : expression->arguments) {
-        const auto jit_expression = _try_translate_expression_to_jit_expression(argument, jit_source, input_node);
+        const auto jit_expression =
+            _try_translate_expression_to_jit_expression(argument, jit_source, input_node, !use_value_ids);
         if (!jit_expression) return nullptr;
         jit_expression_arguments.emplace_back(jit_expression);
       }
 
-      const auto jit_expression_type = _expression_to_jit_expression_type(expression);
+      auto jit_expression_type = _expression_to_jit_expression_type(expression);
 
       if (jit_expression_arguments.size() == 1) {
-        return std::make_shared<JitExpression>(jit_expression_arguments[0], jit_expression_type,
-                                               jit_source.add_temporary_value());
+        const auto jit_expression = std::make_shared<JitExpression>(jit_expression_arguments[0], jit_expression_type,
+                                                                    jit_source.add_temporary_value());
+        if (use_value_ids) {
+          jit_source.add_value_id_expression(jit_expression);
```

Comment:
What happens if the input table uses a different compression? Will the performance be the same as before?